lib/connections/quic_listen.go

```diff
@@ -105,53 +109,33 @@ func (t *quicListener) serve(stop chan struct{}) error {
 	l.Infof("QUIC listener (%v) starting", packetConn.LocalAddr())
 	defer l.Infof("QUIC listener (%v) shutting down", packetConn.LocalAddr())
 
-	// Accept is forever, so handle stops externally.
-	go func() {
-		select {
-		case <-stop:
-			_ = listener.Close()
-		}
-	}()
-
 	for {
-		// Blocks forever, see https://github.com/lucas-clemente/quic-go/issues/1915
-		session, err := listener.Accept()
-
 		select {
 		case <-stop:
-			if err == nil {
-				_ = session.Close()
-			}
 			return nil
 		default:
 		}
+
```

Comment:
Don't need the select thing above as the parent context would be cancelled? Potentially the for loop should check for ctx.Done()'ness?

Reply:
We could check the ctx instead of the stop channel if we like, but we need to break the loop somehow so I think the select is still required as-is.

Reply:
Oh now I see what you mean

Reply:
Or, well, we could `for ctx.Err() == nil {` instead; do you think that's cleaner?

Reply:
Idk, my point is the context is the token of termination, not the channel so we should use that.